guppylang/tys/parsing.py

```diff
@@ -252,8 +256,11 @@ def type_from_ast(
 ) -> Type:
     """Turns an AST expression into a Guppy type."""
     ty, flags = type_with_flags_from_ast(node, globals, param_var_mapping)
-    if flags != InputFlags.NoFlags:
-        raise GuppyError("`@` type annotations are not allowed in this position", node)
+    if flags != InputFlags.NoFlags and flags != InputFlags.Inout:
```

Comment:
Why exclude `InputFlags.Inout` here?

Reply:
If a linear variable has `InputFlags.Inout`, it's because we set it - it can't have been done via an annotation from the user

Reply:
Updated to an assert